File: node/test/api/jdy/form.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/06/07
 */

import { FormApiClient } from '../../../src/api/jdy/form';
import { API_KEY, HOST, APP_ID, ENTRY_ID } from '../../../src/constants/http_constant';

export const formTest = 'formTest';

describe('form api test', () => {
    const api = new FormApiClient(API_KEY, HOST, 'v5');

    test('formWidgets', async () => {
        const formWidgets = await api.formWidgets(APP_ID, ENTRY_ID);
        console.log('formWidgets result:' + JSON.stringify(formWidgets));
        expect(formWidgets['widgets']).toBeTruthy();
        expect(formWidgets['sysWidgets']).toBeTruthy();
        expect(formWidgets['dataModifyTime']).toBeTruthy();
    });
});

File: csharp/test/api/jdy/Form.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("Form")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class FormTest
{

    private static FormApiClient api = new FormApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void Test1()
    {
        var formWidgets = await api.formWidgets(AppTest.appId, AppTest.entryId);
        Assert.True(formWidgets != null);
        Console.WriteLine(formWidgets);
    }
}